src/main/java/reposense/authorship/FileInfoAnalyzer.java

```diff
@@ -77,7 +99,29 @@ private static FileResult generateFileResult(FileInfo fileInfo) {
             Author author = line.getAuthor();
             authorContributionMap.put(author, authorContributionMap.getOrDefault(author, 0) + 1);
         }
-        return new FileResult(fileInfo.getPath(), fileInfo.getFileType(), fileInfo.getLines(), authorContributionMap);
+        return FileResult.createNonBinaryFileResult(
+            fileInfo.getPath(), fileInfo.getFileType(), fileInfo.getLines(), authorContributionMap);
+    }
+
+    /**
+     * Generates and returns a {@code FileResult} with the authorship results from binary {@code fileInfo} consolidated.
+     * Returns null if none of the {@code Author} specified in {@code config} contributed to the file in
+     * {@code fileInfo}.
+     */
+    private static FileResult generateBinaryFileResult(RepoConfiguration config, FileInfo fileInfo) {
+        String authorsString = GitLog.getBinaryFileAuthors(config, fileInfo.getPath());
```

Comment:
It seems like this code gets all the authors that have modified the binary file. 

Are we crediting all the users that have modified the binary file within since date and until date? or should we only get the author that last modified the binary file?

Reply:
Yes it is getting all the authors that have modified the binary file.

I think that we should get all the authors so that it is kind of similar to what we have right now. Currently multiple authors who modified the same non-binary file are all credited.